pkg/create/templates/test.go

```diff
@@ -43,7 +43,19 @@ container_exists() {
 }
 
 container_ip() {
-  docker inspect --format="{{"{{"}} .NetworkSettings.IPAddress {{"}}"}}" $(cat $cid_file)
+  if [ ! -z "$DOCKER_HOST" ] && (echo "$OSTYPE" | grep -qs 'darwin'); then
+    docker-machine ip
+  else
+    docker inspect --format="{{"{{"}} .NetworkSettings.IPAddress {{"}}"}}" $(cat $cid_file)
+  fi
+}
+
+container_port() {
+  if [ ! -z "$DOCKER_HOST" ] && (echo "$OSTYPE" | grep -qs 'darwin'); then
+    docker inspect --format='{{(index (index .NetworkSettings.Ports "8080/tcp") 0).HostPort}}' $(cat $cid_file)
```

Comment:
The `{{` is interpreted by the `template/text` package -- this whole block is a string with a Bash script that gets processed as a template.

Just like in the line we are replacing, we need to "escape" the `{{`. The existing solution involves using the template language to print the `{{`:

```
docker inspect --format='{{"{{"}}(index (index .NetworkSettings.Ports "8080/tcp") 0).HostPort{{"{{"}}' $(cat $cid_file)
```

(Note the `{{"{{"}}` part)

An alternative solution I've seen when a Go template includes a Go template is to [specify a custom delimiter](https://golang.org/pkg/text/template/#Template.Delims), what would require changing the code that processes the templates + updating the templates to use the new delimiters.